src/parser/semantic_check.c

```diff
@@ -9928,6 +9930,121 @@ pt_check_kill (PARSER_CONTEXT * parser, PT_NODE * node)
     }
 }
 
+/*
+ * pt_check_alter_serial () - do semantic checks on the alter serial statement
+ *   return:  none
+ *   parser(in): the parser context used to derive the statement
+ *   node(in): a statement
+ */
+static void
+pt_check_alter_serial (PARSER_CONTEXT * parser, PT_NODE * node)
+{
+  DB_OBJECT *serial_class = NULL, *serial_object = NULL;
+  DB_IDENTIFIER serial_obj_id;
+  DB_OBJECT *owner_object = NULL;
+  PT_NODE *start_val = NULL;
+  PT_NODE *increment_val = NULL;
+  PT_NODE *max_val = NULL;
+  PT_NODE *min_val = NULL;
+  int cyclic;
+  int no_max;
+  int no_min;
+  int no_cyclic;
+  PT_NODE *cached_num_val = NULL;
+  int no_cache;
+  const char *name = NULL;
+  const char *owner_name = NULL;
+
+  OID_SET_NULL (&serial_obj_id);
+
+  assert (node->node_type == PT_ALTER_SERIAL);
+
+  /* find db_serial_class */
+  serial_class = sm_find_class (CT_SERIAL_NAME);
+  if (serial_class == NULL)
+    {
+      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_QPROC_DB_SERIAL_NOT_FOUND, 0);
+      return;
+    }
+
+  /* serial_name */
+  name = node->info.serial.serial_name->info.name.original;
+  if (name == NULL)
+    {
+      assert (sm_check_system_class_by_name (name));
+    }
+
+  /* Check if a serial object exists. */
+  serial_object = do_get_serial_obj_id (&serial_obj_id, serial_class, name);
+  if (serial_object == NULL)
+    {
+      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_QPROC_SERIAL_NOT_FOUND, 1, name);
+      PT_ERRORmf (parser, node, MSGCAT_SET_PARSER_RUNTIME, MSGCAT_RUNTIME_RT_SERIAL_NOT_DEFINED, name);
+      return;
+    }
+
+  assert (node->info.serial.code != NULL);
+  switch (node->info.serial.code)
+    {
+    case PT_SERIAL_OPTION:
+      start_val = node->info.serial.start_val;
+      increment_val = node->info.serial.increment_val;
+      max_val = node->info.serial.max_val;
+      min_val = node->info.serial.min_val;
+      cyclic = node->info.serial.cyclic;
+      no_max = node->info.serial.no_max;
+      no_min = node->info.serial.no_min;
+      no_cyclic = node->info.serial.no_cyclic;
+      cached_num_val = node->info.serial.cached_num_val;
+      no_cache = node->info.serial.no_cache;
+
+      if (!start_val && !increment_val && !max_val && !min_val
+	  && cyclic == 0 && no_max == 0 && no_min == 0
+	  && no_cyclic == 0 && !cached_num_val && no_cache == 0 && node->info.serial.comment == NULL)
```

Comment:
Can cycle and no_cycle be 0 at the same time?

Reply:
This condition checks if there is no option when executing the old ALTER SERIAL statement.

As shown in the example below, if there is no option after ALTER SERIAL, cycle and no_cycle become 0 at the same time.

ex)
CREATE SERIAL cubrid_seq;
ALTER SERIAL cubrid_seq;